url294e6c45f3_Ecdosis_NMergeNew_tgz-pJ8-TestJ8/src/edu/luc/nmerge/Utilities.java

```diff
@@ -0,0 +1,58 @@
+package edu.luc.nmerge;
+
+import java.io.File;
+import java.io.FileInputStream;
+import java.util.Properties;
+
+public class Utilities {
+	/**
+	 * Load a properties file describing the database properties for connecting to
+	 * @param dbConn the name of the properties file (in the class path)
+	 * @return a loaded resource bundle
+	 */
+	public static Properties loadDBProperties( String dbConn )
+	{
+		if ( dbConn == null )
+			return null;
+		else
+		{
+			String wd = System.getProperty("user.dir");
+			File wdDir = new File( wd );
+			Properties props = new Properties();
+	        try
+	        {
+	        	FileInputStream fis = new FileInputStream(wdDir
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code contains a resource that might not be closed properly. This can cause a resource leak that slows down or crashes your system.

**Fix**
Consider closing the following resource: *fis*. The resource is referenced at line 26. The resource is closed at line 27. There are other execution paths that do not contain closure statements, for example, when Properties.load throws an exception. Ensure *fis* is closed in a try-finally block or declared in a try-with-resources block.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).